tests/infrastructure/instance_types/supported_os/conftest.py

```diff
@@ -1,40 +1,29 @@
 import pytest
-from ocp_resources.data_source import DataSource
 
-from utilities.constants import DATA_SOURCE_NAME
-from utilities.storage import data_volume_template_with_source_ref_dict
-from utilities.virt import VirtualMachineForTests
+from tests.infrastructure.instance_types.supported_os.utils import golden_image_vm_with_instance_type
 
 
 @pytest.fixture(scope="class")
-def skip_if_rhel8(instance_type_rhel_os_matrix__module__):
+def xfail_if_rhel8(instance_type_rhel_os_matrix__module__):
     current_rhel_name = [*instance_type_rhel_os_matrix__module__][0]
     if current_rhel_name == "rhel-8":
         pytest.xfail("EFI is not enabled by default before RHEL9")
 
 
 @pytest.fixture(scope="module")
-def golden_image_vm_with_instance_type(
+def golden_image_rhel_vm_with_instance_type(
```

Comment:
if you want to re-use the fixture for other operating systems, you can pass the `os_matrix` fixture name via request.param and then use it
```
os_matirx_fixture = request.param["os_matirx_fixture_name"]
os_matrix = request.getfixturevalue(argname=os_matirx_fixture)

....
os_matrix=os_matrix
```

Reply:
i think we can try what Ruth suggests. generic fixture and request param.

Reply:
add `pytest.mark.usefixtures("instance_type_rhel_os_matrix__module__)` on the class
should do the trick

Reply:
Tried to add the `usefixtures` marker, still doesn't work.
I don't think this is possible because we use several test classes in the same file, and the fixture is something that is created during run-time